src/winetricks

```diff
@@ -3271,11 +3271,13 @@ winetricks_early_wine_arch()
 
 winetricks_detect_gui()
 {
-    if [ -n "$1" ]; then
-        if [ "$1" = "kdialog" ] && test -x "$(command -v kdialog 2>/dev/null)"; then
+    if [ "$1" = "--gui" ] && [ -n "$2" ] ; then
```

Comment:
This can't work as designed; the function is only ever passed one argument.

This PR also breaks --gui=zenity, on a KDE desktop with both zenity and kdialog installed.

In any case, I'm not sure what this is intended to fix. For me, on KDE with both installed:
$ winetricks # launches with kdialog, since XDG_CURRENT_DESKTOP is set to KDE.
$ winetricks --gui=kdialog # launches with kdialog
$ winetricks --gui=zenity # launches with zenity
$ winetricks --gui # launches with zenity (the default)

Reply:
> This can't work as designed; the function is only ever passed one argument.

Thank you, I will resend the PR.

> This PR also breaks --gui=zenity, on a KDE desktop with both zenity and kdialog installed.
> 
> In any case, I'm not sure what this is intended to fix. For me, on KDE with both installed: $ winetricks # launches with kdialog, since XDG_CURRENT_DESKTOP is set to KDE. $ winetricks --gui=kdialog # launches with kdialog $ winetricks --gui=zenity # launches with zenity $ winetricks --gui # launches with zenity (the default)

This PR intended to fix follow errors (when zenity is not installed):

```
$ winetricks --gui
Invalid argument for --gui
Valid options are 'zenity' and 'kdialog'

$ XDG_CURRENT_DESKTOP=KDE winetricks --gui
Invalid argument for --gui
Valid options are 'zenity' and 'kdialog'

```

Reply:
> This can't work as designed; the function is only ever passed one argument.

Dear Austin,
can you rereview the commit?